torch/csrc/jit/python/script_init.cpp

```diff
@@ -1304,8 +1304,34 @@ void initJitScriptBindings(PyObject* module) {
           rcbs.push_back(
               pythonResolver(rcb, classDef.name().name(), classType));
         }
+
+        // Compile getters and setters for properties as regular methods.
+        if (classDef.properties().present()) {
+          for (const auto& prop : classDef.properties().get()) {
+            methodDefs.emplace_back(prop.getter());
+            rcbs.push_back(
+                pythonResolver(rcb, classDef.name().name(), classType));
+            if (prop.setter().present()) {
+              methodDefs.emplace_back(prop.setter().get());
+              rcbs.push_back(
+                  pythonResolver(rcb, classDef.name().name(), classType));
+            }
+          }
+        }
+
         const auto self = SimpleSelf(classType);
         cu->define(classname, methodDefs, rcbs, &self);
+
+        // Add properties onto type definition.
+        if (classDef.properties().present()) {
```

Comment:
I think it would, but is there any way to allow that with the current implementation? To enable something like this, we'd have to declare the methods first, and then define them in a separate step.

Reply:
We already have some code to reorder methods so that `__init__` is compiled first. I think I can do something similar to make sure the properties are compiled first. 

But then we'd have to check for dependencies among the property getters and setters in order to determine the order to compile them in :(((

Reply:
I think that is already handled. That is why all the methodDefs are created up front, and then defined at once. 

If you have a class like:
```
@torch.jit.script
class Properties(object):
    def __init__(self, a: int):
        self.a = a

    def hi(self):
        return self.hello()

    def foo(self) -> int:
        return self.hi()

    def hello(self):
        return 3
```
and reorder the methods it will still compile successfully. 

I think you have to do a similar thing with properties, where in `cu->define` you have to upfront specify what the properties are